olp-cpp-sdk-core/src/client/ApiLookupClientImpl.cpp

```diff
@@ -125,6 +138,20 @@ ApiLookupClient::LookupApiResponse ApiLookupClientImpl::LookupApi(
 CancellationToken ApiLookupClientImpl::LookupApi(
     const std::string& service, const std::string& service_version,
     FetchOptions options, ApiLookupClient::LookupApiCallback callback) {
+  const auto& provider =
+      settings_.api_lookup_settings.catalog_endpoint_provider;
+  if (provider) {
+    auto url = provider(catalog_);
+    if (!url.empty()) {
+      url += "/catalogs/" + catalog_.ToCatalogHRNString();
+      OlpClient result_client;
+      result_client.SetBaseUrl(url);
+      result_client.SetSettings(settings_);
+      callback(result_client);
```

Comment:
Please make sure you write in the ApiLookupClientImpl::LookupApi() doxygen that the callback can be called sync in case the static URL is set. You can imagine the surprise on the users face when his mutex locked twice crashes.